PetAdoption-iOS/PetAdoptionTransportKit/PFPet.swift

```diff
@@ -0,0 +1,112 @@
+        self.size = PFPetSizeType(rawValue: json["size"]["$t"].stringValue)!
+        self.description = json["description"]["$t"].string ?? ""
+        self.status = json["status"]["$t"].string ?? ""
+        self.contact = PFContact(json: json["contact"])
+        
+        // The JSON response for options comes back as either a single dictionary, if there is only one
+        // option, or an array of dictionaries.  This check is required to ensure we are reading the
+        // response correctly in either case.
+        if let optionString = json["options"]["option"]["$t"].string
+        {
+            // TODO: Try to make a function for this since it's done in multiple places
+            if let option = PFPetOptionType(rawValue: optionString)
+            {
+                switch option
+                {
+                    case .altered: self.isSpayed = true
+                    case .housetrained, .housebroken: self.isHouseTrained = true
+                    case .noClaws: self.isDeclawed = true
+                    case .noKids: self.isGoodWithKids = false
+                    case .noDogs: self.isGoodWithDogs = false
+                    case .noCats: self.isGoodWithCats = false
+                    default: print("")
+                }
+            }
+        }
+        else if let options = json["options"]["option"].array
+        {
+            for optionJson in options
+            {
+                if let optionString = optionJson["$t"].string
+                {
+                    if let option = PFPetOptionType(rawValue: optionString)
+                    {
+                        switch option
+                        {
+                            case .altered: self.isSpayed = true
+                            case .housetrained, .housebroken: self.isHouseTrained = true
+                            case .noClaws: self.isDeclawed = true
+                            case .noKids: self.isGoodWithKids = false
+                            case .noDogs: self.isGoodWithDogs = false
+                            case .noCats: self.isGoodWithCats = false
+                            default: print("")
+                        }
+                    }
+
+                }
+            }
+        }
+        
+        // TODO: Try to make these checks part of a separate function, since we're doing it
+        //       in more than one place
+        if let breed = json["breeds"]["breed"]["$t"].string
+        {
+            self.breeds.append(breed)
+        }
+        else if let breeds = json["breed"]["breed"].array
+        {
+            for breedJson in breeds
+            {
+                if let breed = breedJson["$t"].string
+                {
+                    self.breeds.append(breed)
+                }
+            }
+        }
+        
+        for jsonObject in json["media"]["photos"]["photo"]
+        {
+            let photo = PFPhoto(json: jsonObject.1)
+            if photo.size == "x"
```

Comment:
Can we break this out into another function with a method name that explains the validation check?  Right now it's a little confusing to read.

Reply:
Yeah I had written a TODO to take care of that.  This pull request was basically an attempt to get the app pushed out (since Android is already on the Play store, we wanted to get the iOS app out to the App Store as well).  The plan was to refactor these out in the next update.